homeassistant/components/light/nanoleaf_aurora.py

```diff
@@ -39,18 +44,46 @@
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Nanoleaf Aurora device."""
     import nanoleaf
-    host = config.get(CONF_HOST)
-    name = config.get(CONF_NAME)
-    token = config.get(CONF_TOKEN)
+    import nanoleaf.setup
+    if DATA_NANOLEAF_AURORA not in hass.data:
+        hass.data[DATA_NANOLEAF_AURORA] = dict()
+
+    token = ''
+    if discovery_info is not None:
+        host = discovery_info['host']
+        name = discovery_info['hostname']
+        # if device already exists via config, skip discovery setup
+        if host in hass.data[DATA_NANOLEAF_AURORA]:
+            return
+        _LOGGER.info("Discovered a new Aurora: %s", discovery_info)
+        conf = load_json(hass.config.path(CONFIG_FILE))
+        if conf.get(host, {}).get('token'):
+            token = conf[host]['token']
+    elif config is not None:
```

Comment:
This will never be `None`. Setup via discovery will pass an empty dict for config. Change the check to `else:`.